Model/SourcesRepository.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterface;
+use Ampersand\DisableStockReservation\Api\SourcesRepositoryInterface;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterfaceFactory;
+use Ampersand\DisableStockReservation\Model\Sources as SourcesModel;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Magento\InventorySourceSelection\Model\Result\SourceSelectionItem;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository implements SourcesRepositoryInterface
+{
+    /**
+     * @var SourcesInterfaceFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesInterfaceFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourcesConverter $sourcesConverter
+     */
+    public function __construct(
+        SourcesInterfaceFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourcesConverter $sourcesConverter
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourcesConverter = $sourcesConverter;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourcesInterface
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourcesInterface
+    {
+        /** @var SourcesModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        if (!$sourcesModel->getId()) {
+            throw new NoSuchEntityException(__('Source model with the order ID "%1', $orderId));
+        }
+
+        return $sourcesModel;
+    }
+
+    /**
+     * @param string $orderId
+     * @param string $itemSku
+     *
+     * @return SourceSelectionItem|null
+     */
+    public function getSourceItemBySku(string $orderId, string $itemSku): ?SourceSelectionItem
+    {
+        $sourceSelectionItems = $this->sourcesConverter->convertSourcesJsonToSourceSelectionItems(
+            $this->getByOrderId($orderId)->getSources()
+        );
+
+        /** @var SourceSelectionItem $item */
+        foreach ($sourceSelectionItems as $item) {
+            if ($item->getSku() === $itemSku) {
```

Comment:
This should probably throw an exception if the item is not found, in a similar way to the other functions in this repository. You could maybe use `array_filter` to filter the source selection items... Or you could amend this https://github.com/AmpersandHQ/magento2-disable-stock-reservation/pull/6/files#diff-efb3bbef1dda55955c24cd4e4c9a5fc9R76 to set the returned array key as the item SKU. You could then simply do something like:
```php
if (!array_key_exists($itemSku, $sourceSelectionItems)) {
    throw new ...
}

return $sourceSelectionItems[$itemSku];
```